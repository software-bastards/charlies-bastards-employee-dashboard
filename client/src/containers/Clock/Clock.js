
import React, {useState,useEffect} from 'react'

import "../../style/clock.scss"
export default function Clock (){
const [time, setTime] = useState(new Date())

const currentTime =()=>{
    setTime(
        new Date()
    )
}

 useEffect(()=>{
    const timer = setTimeout(() => {
        currentTime()
      }, 1000);
      return () => clearTimeout(timer);})

return(
    <div className="clock-component">
        <h1 >{time.toLocaleTimeString()}</h1>
    </div>
)
}