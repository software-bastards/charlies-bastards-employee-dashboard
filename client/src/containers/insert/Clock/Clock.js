import React, { useState } from "react";
import { useEffect } from "react";
import "../../../style/clock.scss";

function Clock() {
  const [date, setDate] = useState(new Date());
  useEffect(() => {
    var timerID = setInterval(() => tick(), 1000);
    return function cleanup() {
      clearInterval(timerID);
    };
  });

  function tick() {
    setDate(new Date());
  }
  return (
    <div>
      <div className="clock">{date.toLocaleTimeString()}</div>
    </div>
  );
}

export default Clock;
