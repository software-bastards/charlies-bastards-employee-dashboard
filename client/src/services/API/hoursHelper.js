import axios from "axios";

/**
 * @function hoursHelper - select the images from the upload table base on the user ID and the selected month
 * @param {*} token -token provide by JWT
 * @param {*} id - from account table
 * @param {*} month - select by the user
 * @returns {array} - an array of objects
 */

export default async function hoursHelper(token, id, month) {
  const response = await axios.get(
    "/displayhours",

    {
      headers: { Authorization: token },
      params: { id: id, month: month },
    }
  );
  return response;
}
