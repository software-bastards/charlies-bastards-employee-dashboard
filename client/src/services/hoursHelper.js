import axios from "axios";

export default async function hoursHelper(token, id) {
  console.log(token);
  console.log(id);
  const response = await axios.post(
    "http://localhost:5000/displayhours",
    { id },
    {
      headers: { Authorization: "Bearer " + token },
    }
  );
  return response;
}
