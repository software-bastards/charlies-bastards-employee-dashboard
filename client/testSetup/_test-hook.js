import React from "react";
import { shallow, mount } from "enzyme";

export default function testHook(runHook, flushEffects = true) {
  function HookWrapper() {
    const output = runHook();

    return <span output={output} />;
  }

  const wrapperFunc = flushEffects ? mount : shallow;
  const wrapper = wrapperFunc(<HookWrapper />);

  return wrapper.find("span").props().output;
}
