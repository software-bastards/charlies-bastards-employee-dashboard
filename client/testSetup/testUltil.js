import rootReducer from "../src/reducers";
import { createStore } from "redux";

/**
 * @function storeFactory
 * @param {object} initialState  -initial State for the store
 * @returns {Store}- Redux Store
 */

export const storeFactory = (initialState) => {
  return createStore(rootReducer, initialState);
};

/**
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {ShallowWrapper}
 * @param {string} val - Value of data-test atribute for search
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

export const initialState = {
  state: [],
  reducer: {},
};

/**
 * @function expectTruthy - test if any object existes in our component
 * @param {*} receiveObj
 */

export const expectTruthy = (receiveObj) => {
  expect(receiveObj).toBeTruthy() && expect(receiveObj).not.toBeNull();
};
