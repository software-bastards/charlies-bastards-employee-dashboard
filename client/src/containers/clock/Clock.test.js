import React from "react";
import { initialState, expectTruthy } from "../../../testSetup/testUltil";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import EnzymeAdapter from "enzyme-adapter-react-16";
import Enzyme, { render } from "enzyme";
import { cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import Clock from "./Clock";
import hoursHelper from "../../services/hoursHelper";
import { mount } from "enzyme";

afterEach(cleanup);
Enzyme.configure({ adapter: new EnzymeAdapter() });
const state = { data: "2:41:16 PM" };
test("if component renders without an error", () => {
  const historyMock = { push: jest.fn() };
  const mockStore = configureStore();
  const store = mockStore(state);
  const mountComponent = () =>
    mount(
      <Router>
        <Provider store={store}>
          <Clock history={historyMock} />
        </Provider>
      </Router>
    );
  const wrap = mountComponent();
  expect(state.data).toBe("2:41:16 PM");
  expect(wrap).toMatchSnapshot();
});
const mockHistoryPush = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));
