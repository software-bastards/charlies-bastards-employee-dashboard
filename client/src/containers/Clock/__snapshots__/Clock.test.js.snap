// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if component renders without an error 1`] = `ReactWrapper {}`;
